documentation/modules/exploit/windows/http/nscp_authenticated_rce.md

```diff
@@ -0,0 +1,31 @@
+## Description
+
+This module allows an attacker with knowledge of the admin password of NSClient++ 0.5.2.35 to start a privilege reverse shell.
+For this module to work, both web interface of NSClient++ and `ExternalScripts`feature
+should be enabled.
+
+## Installation
+
+A vulnerable version of NSClient++ can be downloaded from [here]https://nsclient.org/download/). Then you can help yourself with this [installation guide](https://docs.nsclient.org/api/rest/) to complete the installation. Don't forget to enable the web interface and the `ExternalScripts` feature to allow the exploit to work.
+
+## Verification Steps
+
+List the steps needed to make sure this thing works
+
+- 1. Start `msfconsole`
+- 2. `use exploit/windows/http/nscp_authenticated_rce`
+- 3. `set RHOST <target_host>`
+- 4. `set PASSWORD <admin_password>` to set the admin password of NSClient++ web interface
+- 5. `check` to check if the targeted NSClient++ is vulnerable
+- 6. `set payload <choose_a_payload>` to set a specific payload to send
+- 7. `run` the module to exploit the vulnerability and start a shell
+
+## Options
+
+**RHOSTS**
+
+Set the target host.
+
```

Comment:
Default options such as RHOSTS should not be defined.

```suggestion
```

Reply:
Okay didn't know that, I thought that the documentation file was a kind of "follow all the steps and this module will work for sure".

Reply:
> 
> 
> Okay didn't know that, I thought that the documentation file was a kind of "follow all the steps and this module will work for sure".

It is, but that's what the numbered list above is for.

Also `rhosts` is implied and we end up with the same text in every module document.

There are also lots of other options that a user might want to set which might be required in some instances, but we don't include in the documentation (`WfsDelay` for laggy connections, `SSL` if the host is behind a non-HTTPS reverse proxy for some reason, etc).

Reply:
Also, the `msfdocs_tidy` utility might be useful. It should warn you if your documentation file contains errors or bad practices.

https://github.com/rapid7/metasploit-framework/blob/master/tools/dev/msftidy_docs.rb

Reply:
> > Okay didn't know that, I thought that the documentation file was a kind of "follow all the steps and this module will work for sure".
> 
> It is, but that's what the numbered list above is for.
> 
> Also `rhosts` is implied and we end up with the same text in every module document.
> 
> There are also lots of other options that a user might want to set which might be required in some instances, but we don't include in the documentation (`WfsDelay` for laggy connections, `SSL` if the host is behind a non-HTTPS reverse proxy for some reason, etc).

Understood !

Reply:
> Also, the `msfdocs_tidy` utility might be useful. It should warn you if your documentation file contains errors or bad practices.
> 
> https://github.com/rapid7/metasploit-framework/blob/master/tools/dev/msftidy_docs.rb

Okay I will add that to my good practice routine 😁 Shouldn't it be added to the list of auto-check on github ?

Reply:
> Okay I will add that to my good practice routine grin Shouldn't it be added to the list of auto-check on github ?

I thought it had been. Here's the relevant check anyway:

https://github.com/rapid7/metasploit-framework/blob/66de4deabf9a403b42cc70f0e94f7d586de3c4e7/tools/dev/msftidy_docs.rb#L247-L250